Sources/Purchasing/IntroEligibilityCalculator.swift

```diff
@@ -36,33 +36,37 @@ class IntroEligibilityCalculator {
         }
         Logger.debug(Strings.customerInfo.checking_intro_eligibility_locally)
 
-        var result = candidateProductIdentifiers.reduce(into: [:]) { resultDict, productId in
-            resultDict[productId] = IntroEligibilityStatus.unknown
-        }
+        var result = candidateProductIdentifiers.dictionaryWithValues { _ in IntroEligibilityStatus.unknown }
         do {
             let receipt = try self.receiptParser.parse(from: receiptData)
             Logger.debug(Strings.customerInfo.checking_intro_eligibility_locally_from_receipt(receipt))
 
+            let activeSubscriptionProductIdentifiers = receipt.activeSubscriptionProductIdentifiers
             let purchasedProductIdsWithIntroOffersOrFreeTrials =
-                receipt.purchasedIntroOfferOrFreeTrialProductIdentifiers()
-
-            let allProductIdentifiers =
-                candidateProductIdentifiers.union(purchasedProductIdsWithIntroOffersOrFreeTrials)
+            receipt.purchasedIntroOfferOrFreeTrialProductIdentifiers
```

Comment:
@aboedo good news: turns out we still needed this. Bad news, not sure how to reconcile both scenarios.

Reply:
lol